Source/WebGPU/WebGPU/ComputePipeline.mm

```diff
@@ -83,6 +83,8 @@ static MTLSize metalSize(auto workgroupSize)
     WGSL::Reflection::Compute computeInformation = std::get<WGSL::Reflection::Compute>(entryPointInformation.typedEntryPoint);
 
     auto function = createFunction(library, entryPointInformation, descriptor.compute.constantCount, descriptor.compute.constants, label);
+    if (!function)
+        return ComputePipeline::createInvalid(*this);
```

Comment:
I think we are supposed to generate an internal error here too, but I can add that in a different PR if you prefer.

Would require this patch to Device.h / .mm and a call to `generateAnInternalError("...")`

```
diff --git a/Source/WebGPU/WebGPU/Device.h b/Source/WebGPU/WebGPU/Device.h
index 50de904ff871..a3a71c7f3df2 100644
--- a/Source/WebGPU/WebGPU/Device.h
+++ b/Source/WebGPU/WebGPU/Device.h
@@ -111,6 +111,7 @@ public:
 
     void generateAValidationError(String&& message);
     void generateAnOutOfMemoryError(String&& message);
+    void generateAnInternalError(String&& message);
 
     Instance& instance() const { return m_adapter->instance(); }
     bool hasUnifiedMemory() const { return m_device.hasUnifiedMemory; }
diff --git a/Source/WebGPU/WebGPU/Device.mm b/Source/WebGPU/WebGPU/Device.mm
index 02742ef83621..201569ec603f 100644
--- a/Source/WebGPU/WebGPU/Device.mm
+++ b/Source/WebGPU/WebGPU/Device.mm
@@ -287,6 +287,25 @@ void Device::generateAnOutOfMemoryError(String&& message)
     }
 }
 
+void Device::generateAnInternalError(String&& message)
+{
+    // https://gpuweb.github.io/gpuweb/#abstract-opdef-generate-an-internal-error
+
+    auto* scope = currentErrorScope(WGPUErrorFilter_Internal);
+
+    if (scope) {
+        if (!scope->error)
+            scope->error = Error { WGPUErrorType_Internal, WTFMove(message) };
+        return;
+    }
+
+    if (m_uncapturedErrorCallback) {
+        instance().scheduleWork([protectedThis = Ref { *this }, message = WTFMove(message)]() mutable {
+            protectedThis->m_uncapturedErrorCallback(WGPUErrorType_Internal, WTFMove(message));
+        });
+    }
+}
+
```

Reply:
I'll let you add that in a separate PR, if that's ok. I'm not sure what is the right error to report here.